scripts/create_deprecated_module.lean

```diff
@@ -0,0 +1,175 @@
+/-
+Copyright (c) 2025 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+--import Mathlib.Init
+import Lean.Elab.Command
+import Mathlib.Tactic.Linter.DeprecatedModule
+-- a comment here
+
+/-!
+#  Create a deprecated module
+
+This file defines the lean script for creating a deprecated module.
+-/
+
+open Lean Elab Command
+
+namespace DeprecatedModule
+
+/--
+`getHeader fname fileContent keepTrailing` takes as input two strings and a `Bool`ean.
+It uses
+* `fname`, as the path of a file (which need not exist);
+* `fileContent`, as the content of `fname` (regardless of whether the file exists and what its
+  content it);
+* `keepTrailing` a boolean.
+
+It returns the content of `fileContent` up to the final import, including trailing whitespace and
+comments if `keepTrailing = true`.
+-/
+def getHeader (fname fileContent : String) (keepTrailing : Bool) : IO String := do
+  let (stx, _) ← Parser.parseHeader (Parser.mkInputContext fileContent fname)
+  let stx := if keepTrailing then stx.raw else stx.raw.unsetTrailing
+  let some substring := stx.getSubstring? | throw <| .userError "No substring: we have a problem!"
+  let upToAllImports : Substring := {substring with startPos := 0}
+  return upToAllImports.toString
+
+/--
+`getHeaderFromFileName fname keepTrailing` is similar to `getHeader`, except that it assumes that
+`fname` is the actual path of a file, and uses `fname`'s content as input to `getHeader`.
+-/
+def getHeaderFromFileName (fname : String) (keepTrailing : Bool) : IO String := do
+  getHeader fname (← IO.FS.readFile fname) keepTrailing
+
+/--
+`mkDeprecation customMessage` returns the formatted syntax
+`deprecated_module "customMessage" (since := "YYYY-MM-DD")`,
+where the message is the input and the date is today's date.
+-/
+def mkDeprecation (customMessage : String := "Auto-generated deprecation") :
+    CommandElabM Format := do
+  let msgStx := if customMessage.isEmpty then none else some <| Syntax.mkStrLit customMessage
+  let dateStx := Syntax.mkStrLit s!"{← Std.Time.PlainDate.now}"
+  let stx ← `(command|deprecated_module $[$msgStx]? (since := $dateStx))
+  liftCoreM <| PrettyPrinter.ppCategory `command stx
+
+/--
+The command `#create_deprecated_modules (n)? (comment)? (write)?` generates deprecated modules.
```

Comment:
do you think "generates module deprecations" reads more nicely? (I don't have a strong preference.)

Reply:
Yes, I like it!